tests/test_dia_object.py

```diff
@@ -144,6 +147,40 @@ def test_dia_source_append_and_update(self):
         dia_obj.update()
         self._compare_dia_object_values(dia_obj, 0, compare_values)
 
+    def test_dia_source_append_different_schema(self):
+        """ Test the appending of a DIASource to a DIAObject where the DIASource
+        has a more complex schema.
+
+        We also test that the update function works properly and that the
+        summary statistics are computed as expected.
+        """
+        single_source = create_test_dia_sources(1)
+        dia_obj = DIAObject(single_source, None)
+        self.assertEqual(dia_obj.id, single_source[0].getId())
```

Comment:
Can you do `self.assertEqual(dia_obj, single_source[0])` to test all the fields?

Reply:
No. I'm testing here that some of the more important values have been copied into the dia_object which is not the same data structure as the source record I access from the source catalog.